src/java/org/apache/cassandra/cql3/constraints/ColumnConstraints.java

```diff
@@ -0,0 +1,200 @@
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import com.google.common.collect.ImmutableSet;
+
+// group of constraints for the column
+public class ColumnConstraints implements ColumnConstraint<ColumnConstraints>
+{
+    public static final Serializer serializer = new Serializer();
+
+    private static final Set<Class<? extends AbstractType>> UNSUPPORTED_TYPES = ImmutableSet.of(MapType.class,
+                                                                                                TupleType.class,
+                                                                                                UserType.class,
+                                                                                                CompositeType.class,
+                                                                                                DynamicCompositeType.class);
+
+
+
+    private final List<ColumnConstraint<?>> constraints;
+    private final int constraintsSize;
+    private final boolean isEmpty;
+
+    public ColumnConstraints(List<ColumnConstraint<?>> constraints)
+    {
+        this.constraints = new ArrayList<>(constraints);
+        // These are catched values to avoid making the calculations every time
+        this.constraintsSize = constraints.size();
+        this.isEmpty = constraints.isEmpty();
+    }
+
+    @Override
+    public IVersionedSerializer<ColumnConstraints> serializer()
+    {
+        return serializer;
+    }
+
+    @Override
+    public void appendCqlTo(CqlBuilder builder)
+    {
+        for (ColumnConstraint<?> constraint : constraints)
+            constraint.appendCqlTo(builder);
+    }
+
+    @Override
+    public void evaluate(Class<? extends AbstractType> valueType, Object columnValue) throws ConstraintViolationException
+    {
+        for (ColumnConstraint<?> constraint : constraints)
+            constraint.evaluate(valueType, columnValue);
+    }
+
+    public List<ColumnConstraint<?>> getConstraints()
+    {
+        return constraints;
+    }
+
+    public boolean isEmpty()
+    {
+        return isEmpty;
+    }
+
+    public int getSize()
+    {
+        return constraintsSize;
+    }
+
+    @Override
+    public void validate(ColumnMetadata columnMetadata) throws InvalidConstraintDefinitionException
+    {
+        if (UNSUPPORTED_TYPES.contains(columnMetadata.type.getClass()))
+            throw new InvalidConstraintDefinitionException("Constraint cannot be defined on column '"
+                                                           + columnMetadata.name + "' with type: "
+                                                           + columnMetadata.type.asCQL3Type());
+
+        for (ColumnConstraint<?> constraint : constraints)
+            constraint.validate(columnMetadata);
+    }
+
+    @Override
+    public ConstraintType getConstraintType()
+    {
+        return ConstraintType.COMPOSED;
+    }
+
+    public static class Noop extends ColumnConstraints
```

Comment:
@bbotella you have this class still public as well as the constructor is public so noting prevents people to create more instance of this.

If you make this class private, then you will not be able to use `ColumnConstraints.Noop.INSTANCE`. It is better if there is `public static final ColumnConstraints NO_OP = new Noop();`. That way, a developer does not know nothing at all about the existence of some "Noop" class. He will always work with an object of type `ColumnConstraints`.

You can also think about moving Noop out of `ColumnConstraints` and put it into `ColumnConstraint` interface as inner static class. I do not understand completely why Noop is meant to be used / tied to ColumnConstraints only.

Reply:
Will move the singleton to the main class. I'll keep it at this level tho. The reason is that we need the isEmpty and the getSize methods, that doesn't make much sense on the main interface.

Reply:
@bbotella please make that Noop class itself private. It is still public.